### PR TITLE
MEPTS-983 | RESUMO MENSAL DE PrEP - MISAU

### PR DESCRIPTION
**Pull request by helderjosue**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/TargetGroupCalculation.java [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    public static TargetGroup of(Concept concept) {
+      HivMetadata hivMetadata = Context.getRegisteredComponents(HivMetadata.class).get(0);
+
+      CommonMetadata commonMetadata = Context.getRegisteredComponents(CommonMetadata.class).get(0);
+
+      if (concept.equals(hivMetadata.getAdolescentsAndYouthAtRiskConcept())) {
+        return ADOLESCENT_AND_YOUTH;
+      } else if (concept.equals(commonMetadata.getPregnantConcept())) {
+        return PREGNANT;
+      } else if (concept.equals(commonMetadata.getBreastfeeding())) {
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/TargetGroupCalculation.java:50]: In this type of structure you could use the early return. removing the else clause the code becomes more readable.

**Review by eliazmuchave — APPROVED** [review]
Looks good.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java [hunk 2]
```diff
@@ -39,6 +33,16 @@
...
 
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.openmrs.module.eptsreports.reporting.calculation.generic.KeyPopulationCalculation.KeyPop.*;
+import static org.openmrs.module.eptsreports.reporting.calculation.generic.KeyPopulationCalculation.TYPE;
+import static org.openmrs.module.eptsreports.reporting.calculation.generic.TargetGroupCalculation.TIPO;
+import static org.openmrs.module.eptsreports.reporting.calculation.generic.TargetGroupCalculation.TargetGroup.*;
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java]: try importing only what is needed but NOT everything from the package, avoid using *

**Review by ningosi — APPROVED** [review]
Looks OK, only wtch the wildcard import in future

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java (hunk 2 above)
Comment (helderjosue) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java]: ok, thanks.

**Pull request merged**